src/sio.ts

```diff
@@ -33,11 +33,234 @@ const DIV_QUOTIENT = 0x070; //  Divider result quotient
+const INTERP0_BASE0 = 0x088; // Read/write access to BASE0 register
+const INTERP0_BASE1 = 0x08c; // Read/write access to BASE1 register
+const INTERP0_BASE2 = 0x090; // Read/write access to BASE2 register
+const INTERP0_POP_LANE0 = 0x094; // Read LANE0 result, and simultaneously write lane results to both accumulators (POP)
+const INTERP0_POP_LANE1 = 0x098; // Read LANE1 result, and simultaneously write lane results to both accumulators (POP)
+const INTERP0_POP_FULL = 0x09c; // Read FULL result, and simultaneously write lane results to both accumulators (POP)
+const INTERP0_PEEK_LANE0 = 0x0a0; // Read LANE0 result, without altering any internal state (PEEK)
+const INTERP0_PEEK_LANE1 = 0x0a4; // Read LANE1 result, without altering any internal state (PEEK)
+const INTERP0_PEEK_FULL = 0x0a8; // Read FULL result, without altering any internal state (PEEK)
+const INTERP0_CTRL_LANE0 = 0x0ac; // Control register for lane 0
+const INTERP0_CTRL_LANE1 = 0x0b0; // Control register for lane 1
+const INTERP0_ACCUM0_ADD = 0x0b4; // Values written here are atomically added to ACCUM0
+const INTERP0_ACCUM1_ADD = 0x0b8; // Values written here are atomically added to ACCUM1
+const INTERP0_BASE_1AND0 = 0x0bc; // On write, the lower 16 bits go to BASE0, upper bits to BASE1 simultaneously
+const INTERP1_ACCUM0 = 0x0c0; // Read/write access to accumulator 0
+const INTERP1_ACCUM1 = 0x0c4; // Read/write access to accumulator 1
+const INTERP1_BASE0 = 0x0c8; // Read/write access to BASE0 register
+const INTERP1_BASE1 = 0x0cc; // Read/write access to BASE1 register
+const INTERP1_BASE2 = 0x0d0; // Read/write access to BASE2 register
+const INTERP1_POP_LANE0 = 0x0d4; // Read LANE0 result, and simultaneously write lane results to both accumulators (POP)
+const INTERP1_POP_LANE1 = 0x0d8; // Read LANE1 result, and simultaneously write lane results to both accumulators (POP)
+const INTERP1_POP_FULL = 0x0dc; // Read FULL result, and simultaneously write lane results to both accumulators (POP)
+const INTERP1_PEEK_LANE0 = 0x0e0; // Read LANE0 result, without altering any internal state (PEEK)
+const INTERP1_PEEK_LANE1 = 0x0e4; // Read LANE1 result, without altering any internal state (PEEK)
+const INTERP1_PEEK_FULL = 0x0e8; // Read FULL result, without altering any internal state (PEEK)
+const INTERP1_CTRL_LANE0 = 0x0ec; // Control register for lane 0
+const INTERP1_CTRL_LANE1 = 0x0f0; // Control register for lane 1
+const INTERP1_ACCUM0_ADD = 0x0f4; // Values written here are atomically added to ACCUM0
+const INTERP1_ACCUM1_ADD = 0x0f8; // Values written here are atomically added to ACCUM1
+const INTERP1_BASE_1AND0 = 0x0fc; // On write, the lower 16 bits go to BASE0, upper bits to BASE1 simultaneously
+
 //SPINLOCK
 const SPINLOCK_ST = 0x5c;
 const SPINLOCK0 = 0x100;
 const SPINLOCK31 = 0x17c;
 
+class InterpolatorConfig {
+  shift = 0;
+  mask_lsb = 0;
+  mask_msb = 0;
+  is_signed = false;
+  cross_input = false;
+  cross_result = false;
+  add_raw = false;
+  force_msb = 0;
```

Comment:
The naming convention is to use `camelCase`, so please change the member names accordingly (e.g. maskLSB, isSigned, etc).

Reply:
sure :)

Reply:
fixed